src/libertem/io/dataset/base/backend.py

```diff
@@ -204,14 +276,30 @@ def _get_tiles_w_copy(self, tiling_scheme, fileset, read_ranges, read_dtype, nat
                     shape=Shape(shape, sig_dims=sig_dims)
                 )
                 data = data.reshape(shape)
+                skip_tile = False
                 self.preprocess(data, tile_slice)
-                yield DataTile(
-                    data,
-                    tile_slice=tile_slice,
-                    scheme_idx=scheme_idx,
-                )
-
-    def get_tiles(self, tiling_scheme, fileset, read_ranges, roi, native_dtype, read_dtype):
+                if sync_offset < 0 and roi is None:
```

Comment:
I'm using `if sync_offset < 0 and roi is None:` to skip tiles or empty some frames of a tile in case of a negative offset without a `roi`.

> What happens in the case of using a roi and having a sync_offset set?

There's no need of skipping tiles or emptying frames in this case. I'll put more comments in my code too

Reply:
Yes, a comment would be good here!